redisinsight/ui/src/services/workbenchStorage.ts

```diff
@@ -0,0 +1,350 @@
+    return {
+      storeName,
+      keyName,
+    }
+  }
+
+  private _normalizeKey(key: unknown) {
+    // Cast the key to a string, as that's all we can set as a key.
+    if (typeof key !== 'string') {
+      console.warn(`${key} used as a key, but it is not a string.`)
+      return String(key)
+    }
+
+    return key
+  }
+}
+
+export const wbHistoryStorage = new WorkbenchStorage('RI_WB_HISTORY', 1)
+
+type CommandHistoryItem = {
+  dbId: string, // Instance Id
+  commandId: string,
+  data: CommandExecution[]
+}
+
+type CommandHistoryType = CommandHistoryItem[]
+
+export async function getLocalWbHistory(instanceId: string) {
+  try {
+    const key = `${BrowserStorageItem.wbCommandsHistory}:${instanceId}`
+    const history = await wbHistoryStorage.getItem(key) as CommandHistoryType
+
+    return history || []
+  } catch (e) {
+    console.error(e)
+    return []
+  }
+}
+
+export function saveLocalWbHistory(commandsHistory: CommandHistoryType) {
+  try {
+    const key = BrowserStorageItem.wbCommandsHistory
+    return Promise.all(commandsHistory.map((chItem) => wbHistoryStorage.setItem(key, chItem)))
+  } catch (e) {
+    console.error(e)
+    return null
+  }
+}
+
+async function cleanupDatabaseHistory(dbId: string) {
+  const commandsHistory: CommandHistoryType = await getLocalWbHistory(dbId)
+  let size = 0
+  const update = commandsHistory.reduce((acc, commandsHistoryElement) => {
+    if (size > riConfig.workbench.maxItemsPerDb) {
+      return acc
+    }
+    let items = commandsHistoryElement.data
+    size += items.length
+    if (size > riConfig.workbench.maxItemsPerDb) {
+      const diff = riConfig.workbench.maxItemsPerDb - size
+      items = items.slice(diff)
+      acc.push({
+        dbId,
+        data: items,
+        commandId: commandsHistoryElement.commandId
+      })
+    }
+
+    return acc
+  }, [] as CommandHistoryType)
+
+  await saveLocalWbHistory(update)
+}
+
+export async function addCommands(dbId: string, commandId: string, data: CommandExecution[]) {
+  // Store command results in local storage!
+  const commandsHistory: CommandHistoryType = await getLocalWbHistory(dbId)
+  const storedData = data.map((item) => {
+    // Do not store command execution result that exceeded limitation
+    if (JSON.stringify(item.result).length > riConfig.workbench.maxResultSize) {
+      item.result = [
+        {
+          status: CommandExecutionStatus.Success,
+          response: `Results have been deleted since they exceed ${formatBytes(riConfig.workbench.maxResultSize)}. 
+          Re-run the command to see new results.`,
+        },
+      ]
+    }
+    return item
+  })
+  await saveLocalWbHistory([
+    ...commandsHistory,
+    {
+      dbId,
+      commandId,
+      data: reverse(storedData),
+    } as CommandHistoryItem
+  ])
+
+  return cleanupDatabaseHistory(dbId)
+}
+
+export async function removeCommand(dbId: string, commandId: string) {
+  // Delete command from local storage?!
```

Comment:
![Uploading deletewb.png…]()

After deleting a workbench item, it doesn't get removed from local storage, so it re-appears after navigating away from the page and coming back. It seems like we should delete it